devel/smake/Portfile

```diff
@@ -1,41 +1,96 @@
 
-checksums           rmd160  f683a4051273b3e24e45a2b435ddb23509db55bf \
-                    sha256  27566aa731a400c791cd95361cc755288b44ff659fa879933d4ea35d052259d4
+name                    smake
+version                 1.7-${codeberg.version}
+revision                0
+categories              devel
+platforms               darwin
+maintainers             {hotmail.com:amtor @RobK88} \
+                        openmaintainer
+license                 CDDL
+description             A highly portable make program with automake features
+long_description        Smake is a highly portable make program with automake \
+                        features. It is currently mainly targeted to be used \
+                        with the Schily SING makefile system.
 
-use_configure	no
+homepage                https://codeberg.org/schilytools/schilytools
+master_sites            ${homepage}/archive/
+checksums               rmd160  e154278ecbe7d778bc1d6766ed163c9963b1cc82 \
+                        sha256  a4270cdcca5dd69c0114079277b06e5efad260b0a099c9c09d31e16e99a23ff5 \
+                        size    5896292
 
-# https://trac.macports.org/ticket/31616
-compiler.blacklist-append llvm-gcc-4.2 {clang < 318}
-
-build.type		gnu
-build.args      CC=${configure.cc}
-use_parallel_build  no
-
-destroot.args       CC=${configure.cc}
-destroot.destdir	INS_BASE=${destroot}${prefix} \
-					MANDIR=man \
-					INSUSR=`id -u` INSGRP=`id -g`
-post-destroot	{
-	file delete -force ${destroot}${prefix}/share/man/man5 \
-		${destroot}${prefix}/lib ${destroot}${prefix}/include
+post-extract {
+                        copy ${filespath}/Gmake.smake ${worksrcpath}
 }
 
-livecheck.url       https://sourceforge.net/projects/s-make/files/
-livecheck.regex     ${name}-(\\d+(?:\\.\\d+)*)${extract.suffix}
+patchfiles              001-Makefile-Patch.diff
+
+post-patch {
+                        reinplace -locale C "s|/opt/schily|${prefix}|g" \
+                        ${worksrcpath}/DEFAULTS/Defaults.darwin \
+                        ${worksrcpath}/DEFAULTS/Defaults.mac-os10 \
+                        ${worksrcpath}/DEFAULTS_ENG/Defaults.darwin \
+                        ${worksrcpath}/DEFAULTS_ENG/Defaults.mac-os10 \
+                        ${worksrcpath}/libfind/find.c \
+                        ${worksrcpath}/libfind/find_main.c \
+                        ${worksrcpath}/librscg/scsi-remote.c \
+                        ${worksrcpath}/patch/tests/random/cmptest.sh \
+                        ${worksrcpath}/patch/tests/random/gentest.sh \
+                        ${worksrcpath}/smake/job.c \
+                        ${worksrcpath}/smake/make.c \
+                        ${worksrcpath}/smake/smake.1 \
+                        ${worksrcpath}/TEMPLATES/Defaults.gcc \
+                        ${worksrcpath}/TEMPLATES/Defaults.clang \
+                        ${worksrcpath}/TEMPLATES/Defaults.xcc
+}
+
+conflicts_build-append  cdrtools
```

Comment:
> I suggest to add a note / comment with instruction why such conflicts is here, and try when it safe to be removed :)

Agreed!

Reply:
Since the problem ONLY occurs with OLD versions of `cdrtools`, there is a MUCH better way to deal with this issue, if someone could tell me how to get the version number of the active port from the MacPorts registry.  

Here is my draft code.  I need help with  the second `if` statement.  I am sure there is a way to get the version number of the active port from the MacPorts registry.  I just do not know how and there is almost no docs that I can find on the MacPorts registry.

```
pre-configure{

     if {![catch {registry_active cdrtools} result]} {
                set regentry [lindex $result 0]
                if [ activeport.version =< 3.01} {
                          conflicts.build    cdrtools
             }
     }
}
```

Reply:
This is the reusable code I use, when a conflict check is necessary:

```
proc port_conflict_check {p_port_name p_conflict_ver_min p_conflict_ver_max} {
    ui_info "Checking for conflict against port: ${p_port_name}"

    if { ![catch {set port_conflict_ver_info [lindex [registry_active ${p_port_name}] 0]}] } {
        set port_conflict_ver [lindex ${port_conflict_ver_info} 1]_[lindex ${port_conflict_ver_info} 2]
        ui_info "${p_port_name} active version: ${port_conflict_ver}"

        if { [vercmp ${port_conflict_ver} ${p_conflict_ver_min}] >= 0
            && [vercmp ${port_conflict_ver} ${p_conflict_ver_max}] <= 0 } { 

            ui_info "${p_port_name} conflicts; declare build conflict"
            conflicts_build-append \
                ${p_port_name}
        }
    }
}

port_conflict_check cdrtools 0.0 3.01
```

To see the output from the proc, run `port -v info smake`.

Reply:
> @mascguy maybe add your code to `conflicts_build` PG?

Yep, definitely!

Reply:
> @macsguy - Fantastic! Where does one normally place the proc subroutine code in the Portfile? In the case of this Portflile for `smake` , inside the `pre-configure {}` block? See below. Or do we even need the `pre-configure {}` code block at all? Probably not but it does make the Portfile easier to read and understand.

Place the proc - and the call to it - at the top level of the portfile. (Typically I place procs toward the end of the port, though that's totally up to you.)